### DISCUSSION TITLE
NEO Coolcam NAS-WR01W 16A Power Monitoring Plug

### DISCUSSION THREAD

**Discussion by emilof** [RavenSystem/esp-homekit-devices#1842] [MEPLHAA Scripts Collection]
**Device Name:** NEO Coolcam NAS-WR01W 16A Power Monitoring Plug
**URL to Device:** https://www.aliexpress.com/item/32966183521.html
**Used HAA version:** 12.9.0
**Summary of functions:** On/off and power monitoring
**MEPLHAA script:**
```json
{
  "c": {
    "io": [
      [[13, 14], 2],
      [[0], 6]
    ],
    "b": [[0, 5]]
  },
  "a": [
    {
      "t": 2,
      "0": {
        "r": [
          [14, 0],
          [13, 1]
        ]
      },
      "1": {
        "r": [
          [14, 1],
          [13, 0]
        ]
      },
      "b": [[0, 1]],
      "es": [
        {
          "t": 75,
          "vf": 0.13334,
          "cf": 0.01295,
          "pf": 1.5312,
          "dt": [4, 5, 12]
        }
      ]
    }
  ]
}
```
Old MEPLHAA script: 
`{"c":{"b":[{"g":0,"t":5}]},"a":[{"0":{"r":[{"g":14},{"g":13,"v":1}]},"1":{"r":[{"g":14,"v":1},{"g":13}]},"b":[{"g":0}],"es":[{"vf":0.13334,"cf":0.01295,"pf":1.5312,"dt":[4,5,12],"t":75}],"t":2}]}`
(Tested with HAA version: 10.7.1 --> 11.2)

GPIO00 Button1
GPIO04 BL0937 CF
GPIO05 HLWBL CF1
GPIO12 HLWBL SELi
GPIO13 Led1i
GPIO14 Relay1
(https://templates.blakadder.com/neo_coolcam_NAS-WR01W-16A.html)

I had some trouble trying to flash this device. After a lot of tinkering and resoldering I finally got it to work by also connecting RST to 3V3 and skip the cross connect on RX/TX. I used a CH340G USB-Serial adapter when flashing.

```
Board    USB Adapter
3V3 ---- 3V3
GND ---- GND
RX ----- RX *
TX ----- TX *
RES ---- 3V3
```
\* No cross connection!

These sources were very helpful when trying to figure out how to open, solder and flash these smart plugs:
- https://getsmarty.tech/posts/easiest-way-to-flash-your-sonoff-shelly-wemos-or-any-esp8266
- http://www.christoph-koenig.de/flash/
- https://www.youtube.com/watch?v=CV8Y61gEQpQ
- https://forum-supla-org.translate.goog/viewtopic.php?t=4407&start=40&_x_tr_sl=auto&_x_tr_tl=en&_x_tr_hl=sv&_x_tr_pto=wapp (Google translated)

**Comment by RavenSystem** [comment]
Many thanks! I ordered two units.

**Comment by branda92** [comment]
I also just took 2 pieces. Thank you!

**Comment by branda92** [comment]
I flashed like any other device. I connected GND-GND, VCC-VCC, TX-RX, RX-TX. I then pressed the button and then released, while turning on to enter the flash mode.
